src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorApiMessageReader.java

```diff
@@ -716,7 +716,15 @@ public static void installFirmware_515(
               + ") from "
               + firmwareUrl);
       try {
-        downloadFirmwareFile(firmwareUrl);
+        // Perform firmware download and store result if the download/update was unsuccessful
+        String result = downloadFirmwareFile(firmwareUrl);
+
+        // Update state to failed with result (device reboots instead of returning when successful)
```

Comment:
This comment makes it sound like the previous method only returns for failure.

Reply:
It does only return on failure. If the firmware update was successful, the device halts the JVM and it never returns.